app/coffeescripts/views/conferences/ConferenceView.coffee

```diff
@@ -126,3 +128,63 @@ define [
         else
           window.open(data[0].url)
       )
+
+    delete_recording: (e) ->
+      return if !confirm(I18n.t('recordings.confirm.delete', "Are you sure you want to delete this recording?"))
+      e.preventDefault()
+      parent = $(e.currentTarget).parent()
+      params = {recording_id: parent.data("id")}
+      this.toggleActionButton(parent, {state: "processing", action: "delete"})
+      this.toggleRecordingLink(parent, {state: "processing"})
+      $.ajaxJSON(parent.data('url') + "/delete_recording", "POST", params,
+        (data) =>
+          if data.deleted
+            this.removeRecordingRow(parent)
+            return
+          this.ensure_delete_performed({attempt: 1, parent: parent, desired_state: "true"})
+      )
+
+    ensure_delete_performed: (payload) ->
+      $.ajaxJSON(payload.parent.data('url') + "/recording", "POST", {
+          recording_id: payload.parent.data("id"),
+        }, (data) =>
+          current_state = if $.isEmptyObject(data) then "true" else "false"
+          if current_state == payload.desired_state
```

Comment:
this should just collapse into `if $.isEmptyObject(data)`

Reply:
Indeed, current_state is actually not required here. Same reason as befor. The generic method in order to avoid duplication :)